src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -221,6 +228,73 @@ protected Object decode(
                                 case 13:
                                     position.setSpeed(UnitsConverter.knotsFromKph(buf.readFloatLE()));
                                     break;
+                                case 14:
+                                    position.setCourse(buf.readUnsignedShortLE());
+                                    break;
+                                case 15:
+                                    position.set(Position.KEY_ODOMETER, buf.readFloatLE());
+                                    break;
+                                case 19:
+                                    position.set(Position.KEY_POWER, buf.readShortLE() * 0.001);
+                                    break;
+                                case 20:
+                                    position.set(Position.KEY_BATTERY, buf.readShortLE() * 0.001);
+                                    break;
+                                case 21:
+                                case 22:
+                                case 23:
+                                case 24:
+                                case 25:
+                                case 26:
+                                    position.set(Position.PREFIX_ADC + (j + 2 - 21), buf.readUnsignedShortLE() * 0.001);
+                                    break;
+                                case 29:
+                                    value = buf.readUnsignedByte();
+                                    for (int k = 0; k <= 7; k++) {
+                                        position.set(Position.PREFIX_IN + (k + 1), BitUtil.check(value, k) ? 1 : 0);
+                                    }
+                                    break;
+                                case 31:
+                                    value = buf.readUnsignedByte();
+                                    for (int k = 0; k <= 3; k++) {
+                                        position.set(Position.PREFIX_OUT + (k + 1), BitUtil.check(value, k) ? 1 : 0);
+                                    }
+                                    break;
+                                case 33:
+                                case 34:
+                                    position.set(Position.PREFIX_COUNT + (j + 2 - 33), buf.readUnsignedIntLE());
+                                    break;
+                                case 35:
+                                case 36:
+                                    position.set("freq" + (j + 2 - 35), buf.readUnsignedShortLE());
+                                    break;
+                                case 37:
+                                    position.set("engine_hours", buf.readUnsignedIntLE());
```

Comment:
Sorry, just noticed this one. You should use `Position.KEY_HOURS` here.

Reply:
Yes, I saw this key, but let's explain why I chose this option and then finally decide which is better...

The devices have 3 engine hours parameters:
1. engine hours from the device according to the ignition sensor (main parameter)
2. engine hours from the device under load (some other mechanism is working)
3. engine hours from CAN-bus

And as in the case of fuel, I thought to use different names.

Maybe for options 1 and 2:
`position.set(Position.KEY_HOURS + xxx, buf.readUnsignedIntLE());`

And for CAN do:
`position.set("canHours", buf.readUnsignedIntLE());`

Reply:
How about `obdHours`? It's going to be consistent with `obdSpeed` etc that we already have.

Reply:
In total, we get...

Now committing KEY_HOURS. And in the future, when adding CAN parameters, I will add the "obd" prefix to them, if I do not find an unambiguous key from the existing ones.

Reply:
ok